clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/loadbalancer/DeleteGoogleInternalLoadBalancerAtomicOperation.groovy

```
@@ -0,0 +1,161 @@
+    def region = description.region
+    def forwardingRuleName = description.loadBalancerName
+
+    task.updateStatus BASE_PHASE, "Retrieving forwarding rule $forwardingRuleName in $region..."
+
+    ForwardingRule forwardingRule =
+      compute.forwardingRules().get(project, region, forwardingRuleName).execute()
+    if (forwardingRule == null) {
+      GCEUtil.updateStatusAndThrowNotFoundException("Forwarding rule $forwardingRuleName not found in $region for $project",
+        task, BASE_PHASE)
+    }
+
+    def backendServiceName = GCEUtil.getLocalName(forwardingRule.backendService)
+
+    task.updateStatus BASE_PHASE, "Retrieving backend service $backendServiceName in $region..."
+
+    BackendService backendService = compute.regionBackendServices().get(project, region, backendServiceName).execute()
+    if (backendService == null) {
+      GCEUtil.updateStatusAndThrowNotFoundException("Backend service $backendServiceName not found in $region for $project",
+        task, BASE_PHASE)
+    }
+
+    def healthCheckUrl = backendService.healthChecks[0]
+    def healthCheckName = GCEUtil.getLocalName(healthCheckUrl)
+    def healthCheckType = Utils.getHealthCheckType(healthCheckUrl)
+    def healthCheck = null
+    switch (healthCheckType) {
+      case "httpHealthChecks":
+        healthCheck = compute.httpHealthChecks().get(project, healthCheckName).execute()
+        break
+      case "httpsHealthChecks":
+        healthCheck = compute.httpsHealthChecks().get(project, healthCheckName).execute()
+        break
+      case "healthChecks":
+        healthCheck = compute.healthChecks().get(project, healthCheckName).execute()
+        break
+      default:
+        log.warn("Unknown health check type for health check named: ${healthCheckName}. Not queueing any batch requests.")
+        break
+    }
+    if (healthCheck == null) {
+      GCEUtil.updateStatusAndThrowNotFoundException("Health check $healthCheckName not found for $project",
```

I throw an exception here instead of just warning.